### ISSUE TITLE
poolmgr fails while creating, invoking and deleting function continually

### ISSUE THREAD

**Issue by yqf3139** [fission/fission#139]
Use the following script to create a temp function, invoke it's default http trigger and then delete it continually.

```
COUNTER=1
while true;
do
fission fn create --name tmp$COUNTER --env python --code hello.py
sleep 4
curl 127.0.0.1:31314/fission-function/tmp$COUNTER
fission fn delete --name tmp$COUNTER
let "COUNTER++"
done

```
In a fresh minikube, with fresh poolmgr and router, the first 12 test is successful, but errors occurred from the 13th test.

```
function 'tmp8' deleted
function 'tmp9' created
Hi
function 'tmp9' deleted
function 'tmp10' created
Hi
function 'tmp10' deleted
function 'tmp11' created
Hi
function 'tmp11' deleted
function 'tmp12' created
Hi
function 'tmp12' deleted
function 'tmp13' created
Internal server error (fission)
function 'tmp13' deleted
function 'tmp14' created
Internal server error (fission)
function 'tmp14' deleted
function 'tmp15' created
Internal server error (fission)
function 'tmp15' deleted
function 'tmp16' created
```

router logs:
```
2017-03-04T06:59:02.308183246Z 2017/03/04 06:59:02 Proxying request for /fission-function/tmp8 to 172.17.0.25:8888
2017-03-04T06:59:02.309984150Z 172.17.0.1 - - [04/Mar/2017:06:59:02 +0000] "GET /fission-function/tmp8 HTTP/1.1" 200 3
2017-03-04T06:59:06.339363047Z 2017/03/04 06:59:06 Not cached, getting new service for {tmp9 26c00d74-7338-4e49-bf00-fd9587128b44}
2017-03-04T06:59:06.384710543Z 2017/03/04 06:59:06 Proxying request for /fission-function/tmp9 to 172.17.0.27:8888
2017-03-04T06:59:06.387306939Z 172.17.0.1 - - [04/Mar/2017:06:59:06 +0000] "GET /fission-function/tmp9 HTTP/1.1" 200 3
2017-03-04T06:59:10.449102423Z 2017/03/04 06:59:10 Not cached, getting new service for {tmp10 ccfaac47-5bdf-4880-8647-defad6b41b23}
2017-03-04T06:59:10.538835410Z 2017/03/04 06:59:10 Proxying request for /fission-function/tmp10 to 172.17.0.28:8888
2017-03-04T06:59:10.543056701Z 172.17.0.1 - - [04/Mar/2017:06:59:10 +0000] "GET /fission-function/tmp10 HTTP/1.1" 200 3
2017-03-04T06:59:14.840749471Z 2017/03/04 06:59:14 Not cached, getting new service for {tmp11 eb0e96dc-70cd-409c-8367-c770c2b10815}
2017-03-04T06:59:15.576506320Z 2017/03/04 06:59:15 Request delay for http://172.17.0.15:8888: 747.215949ms
2017-03-04T06:59:15.576533440Z 2017/03/04 06:59:15 Proxying request for /fission-function/tmp11 to 172.17.0.15:8888
2017-03-04T06:59:15.579598357Z 172.17.0.1 - - [04/Mar/2017:06:59:14 +0000] "GET /fission-function/tmp11 HTTP/1.1" 200 3
2017-03-04T06:59:19.637786172Z 2017/03/04 06:59:19 Not cached, getting new service for {tmp12 e02d1a05-353d-4069-b7d2-7425ea46cd45}
2017-03-04T06:59:20.096463228Z 2017/03/04 06:59:20 Request delay for http://172.17.0.14:8888: 481.843023ms
2017-03-04T06:59:20.096463747Z 2017/03/04 06:59:20 Proxying request for /fission-function/tmp12 to 172.17.0.14:8888
2017-03-04T06:59:20.097186138Z 172.17.0.1 - - [04/Mar/2017:06:59:19 +0000] "GET /fission-function/tmp12 HTTP/1.1" 200 3
2017-03-04T06:59:24.143896696Z 2017/03/04 06:59:24 Not cached, getting new service for {tmp13 115b93fd-555e-420a-bbd1-3e94f6e61004}
2017-03-04T06:59:24.712975386Z 2017/03/04 06:59:24 Failed to get service for function (tmp13,115b93fd-555e-420a-bbd1-3e94f6e61004): Internal error - Pod has no IP
2017-03-04T06:59:24.713005206Z 172.17.0.1 - - [04/Mar/2017:06:59:24 +0000] "GET /fission-function/tmp13 HTTP/1.1" 500 32
2017-03-04T06:59:28.821215426Z 2017/03/04 06:59:28 Not cached, getting new service for {tmp14 db8f3087-a39a-480f-bb1e-48a19351ee3e}
2017-03-04T06:59:29.078575179Z 172.17.0.1 - - [04/Mar/2017:06:59:28 +0000] "GET /fission-function/tmp14 HTTP/1.1" 500 32
2017-03-04T06:59:29.091157058Z 2017/03/04 06:59:29 Failed to get service for function (tmp14,db8f3087-a39a-480f-bb1e-48a19351ee3e): Internal error - Pod has no IP
2017-03-04T06:59:33.238382862Z 2017/03/04 06:59:33 Not cached, getting new service for {tmp15 5901b15c-db81-4a02-9d77-011799028c2a}
2017-03-04T06:59:34.985308978Z 172.17.0.1 - - [04/Mar/2017:06:59:33 +0000] "GET /fission-function/tmp15 HTTP/1.1" 500 32
2017-03-04T06:59:34.985390307Z 2017/03/04 06:59:34 Failed to get service for function (tmp15,5901b15c-db81-4a02-9d77-011799028c2a): Internal error - Pod has no IP
```

poolmgr logs (error part):
```
2017-03-04T06:59:28.977273310Z 2017/03/04 06:59:28 [map[poolmgrInstanceId:fMIJGGKz unmanaged:true functionName:tmp14 functionUid:db8f3087-a39a-480f-bb1e-48a19351ee3e]] found 2 ready pods of 3 total
2017-03-04T06:59:28.977578364Z 2017/03/04 06:59:28 relabeling pod: [python-458acd7d-51df-4f4e-93a0-f7a441fa855b-7sguevlk-682588sl0j]
2017-03-04T06:59:29.077237829Z 2017/03/04 06:59:29 Chosen pod: python-458acd7d-51df-4f4e-93a0-f7a441fa855b-7sguevlk-682588sl0j (in 113.914303ms)
2017-03-04T06:59:29.077318910Z 2017/03/04 06:59:29 Error: 500: Pod has no IP
2017-03-04T06:59:29.078340943Z 172.17.0.7 - - [04/Mar/2017:06:59:28 +0000] "POST /v1/getServiceForFunction HTTP/1.1" 500 14
2017-03-04T06:59:29.098306755Z 2017/03/04 06:59:29 Error in pod 'python-458acd7d-51df-4f4e-93a0-f7a441fa855b-7sguevlk-682588sl0j', scheduling cleanup
2017-03-04T06:59:33.238540878Z 2017/03/04 06:59:33 [tmp15] Checking for cached function service
2017-03-04T06:59:33.238567620Z 2017/03/04 06:59:33 [tmp15] No cached function service found, creating one
2017-03-04T06:59:33.238579165Z 2017/03/04 06:59:33 [tmp15] getting environment for function
2017-03-04T06:59:33.238589162Z 2017/03/04 06:59:33 [&{tmp15 5901b15c-db81-4a02-9d77-011799028c2a}] getting function from controller
2017-03-04T06:59:33.238598280Z 2017/03/04 06:59:33 [&{tmp15 5901b15c-db81-4a02-9d77-011799028c2a}] getting env from controller
2017-03-04T06:59:33.241199008Z 2017/03/04 06:59:33 [tmp15] getting generic pool for env
2017-03-04T06:59:33.241223742Z 2017/03/04 06:59:33 [tmp15] getting function service from pool
2017-03-04T06:59:33.241262773Z 2017/03/04 06:59:33 [&{tmp15 5901b15c-db81-4a02-9d77-011799028c2a}] Choosing pod from pool
2017-03-04T06:59:33.444139294Z 2017/03/04 06:59:33 [map[poolmgrInstanceId:fMIJGGKz unmanaged:true functionName:tmp15 functionUid:5901b15c-db81-4a02-9d77-011799028c2a]] found 2 ready pods of 3 total
2017-03-04T06:59:33.444234775Z 2017/03/04 06:59:33 relabeling pod: [python-458acd7d-51df-4f4e-93a0-f7a441fa855b-7sguevlk-682581m48m]
2017-03-04T06:59:34.299354038Z 2017/03/04 06:59:34 failed to relabel pod [python-458acd7d-51df-4f4e-93a0-f7a441fa855b-7sguevlk-682581m48m]: Operation cannot be fulfilled on pods "python-458acd7d-51df-4f4e-93a0-f7a441fa855b-7sguevlk-682581m48m": the object has been modified; please apply your changes to the latest version and try again
2017-03-04T06:59:34.839745522Z 2017/03/04 06:59:34 [map[unmanaged:true functionName:tmp15 functionUid:5901b15c-db81-4a02-9d77-011799028c2a poolmgrInstanceId:fMIJGGKz]] found 2 ready pods of 3 total
2017-03-04T06:59:34.841376222Z 2017/03/04 06:59:34 relabeling pod: [python-458acd7d-51df-4f4e-93a0-f7a441fa855b-7sguevlk-682581m48m]
2017-03-04T06:59:34.966803540Z 172.17.0.7 - - [04/Mar/2017:06:59:33 +0000] "POST /v1/getServiceForFunction HTTP/1.1" 500 14
2017-03-04T06:59:34.966922458Z 2017/03/04 06:59:34 Chosen pod: python-458acd7d-51df-4f4e-93a0-f7a441fa855b-7sguevlk-682581m48m (in 1.725898319s)
2017-03-04T06:59:34.966937733Z 2017/03/04 06:59:34 Error: 500: Pod has no IP
2017-03-04T06:59:34.966946752Z 2017/03/04 06:59:34 Error in pod 'python-458acd7d-51df-4f4e-93a0-f7a441fa855b-7sguevlk-682581m48m', scheduling cleanup
```

Wait for a while, the above situation can be reproduced. (6 success and following some errors and success)

#88

**Comment by yqf3139** [comment]
In a normal k8s, this error is fixed by #141 .
However in minikube, there is a limitation of pods and causes poormgr and router crashed and restarted.

**Comment by soamvasani** [comment]
Let's keep this open till the fix is merged.

As for reaching pod limits: testing is a good use case for pod reuse: #54
